ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/concatenate/AlterTableConcatenateOperation.java

```diff
@@ -76,7 +76,9 @@ private MergeFileWork getMergeFileWork(CompilationOpContext opContext) throws Hi
 
     Map<Path, List<String>> pathToAliases = new LinkedHashMap<>();
     List<String> inputDirStr = Lists.newArrayList(inputDirList.toString());
-    pathToAliases.put(desc.getInputDir(), inputDirStr);
+    for (Path path: mergeWork.getInputPaths()) {
```

Comment:
@ngsg, `mergeWork.getInputPaths()` and `desc.getInputDir()` are the same, isn't it?
````
mergeWork = new MergeFileWork(Lists.newArrayList(desc.getInputDir())
````